File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int n = people.length;
        int i = 0, j = n - 1, cnt = 0;
        while (i <= j) {
            if (people[j] + people[i] <= limit) {
                i++;
                j--;
            } else {
                j--;
            }
            cnt++;
        }
        return cnt;
    }
}

File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        
        int i = 0, j = people.size() - 1, boats = 0;
        
        while (i < j) {
            
            if (people[i] + people[j] <= limit) {
                boats++;
                i++;
                j--;
            } else {
                boats++;
                j--;
            }
            
        }
        
        if (i == j) {
            boats++;
        }
        
        return boats;
    }
};